protoc-gen-twirp/generator.go

```diff
@@ -1146,6 +1146,21 @@ func (t *twirp) generateServer(file *descriptor.FileDescriptorProto, service *de
 	t.P(`}`)
 	t.P()
 
+	// Write request body read issue
+	t.P(`// handleRequestBodyError is used to handle error when the twirp server cannot read request`)
+	t.P(`func (s *`, servStruct, `) handleRequestBodyError(ctx `, t.pkgs["context"], `.Context, resp `, t.pkgs["http"], `.ResponseWriter, msg string, err error) {`)
+	t.P(`  if ctxErr := context.Canceled; ctxErr == ctx.Err() {`)
+	t.P(`    s.writeError(ctx, resp, twirp.NewError(twirp.Canceled, ctxErr.Error()))`)
```

Comment:
It looks like all references to imported packages need to go via the `t.pkgs` map, as they do on the line above. That would hold for the `context` and `twirp` references here and in the next stanza, and for the following `twirp` reference.

(It's unfortunate that we don't have tests to detect that; I don't think it could break based on user input (the name of the protobuf package, for example) since "our" calls to `registerPackageName` come so early, so I'd be surprised if this would be the only part of the generator that skipped the map.)

Reply:
Nice catch! and good point about the tests, thanks Rhysh

Reply:
Great catch :)
Fixed.